src/App.js

```diff
@@ -25,7 +25,7 @@ const App = () => {
       <div className="App">
         <Switch>
           <Route exact path="/">
-            <Home setToken={setToken} />
+            {token ? <Redirect to="/list" /> : <Home setToken={setToken} />}
           </Route>
           <Route exact path="/list">
             {token ? <List token={token} /> : <Redirect to="/" />}
```

Comment:
Because we are passing down our `token` prop. Why don't we go ahead and pass it into our `useCollection()` function in the `List.js` component?

Something like:

```javascript
 const [listItem, loading, error] = useCollection(db.collection(props.token), {
    snapshotListenOptions: { includeMetadataChanges: true },
  });
```

Which allows us to see whatever items are already there (if there are any) instead of always seeing `generated_token`.

Reply:
That's a great idea! :D